src/Confluent.Kafka/Consumer.cs

```diff
@@ -195,12 +188,13 @@ public void Poll(TimeSpan timeout)
 
         /// <summary>
         ///     Poll for new consumer events, including new messages
-        ///     ready to be consumed(which will trigger the OnMessage
+        ///     ready to be consumed (which will trigger the OnMessage
         ///     event).
         /// </summary> 
         /// <remarks>
         ///     Blocks indefinitely until a new event is ready.
         /// </remarks>
+        [Obsolete("Indefinite polling cannot be cancelled so it's usually not what you want. Removing this method to discourage.", false)]
```

Comment:
Just "Use Poll overload with finite timeout" is enough I think (obsolete is already discouraging)